### DISCUSSION TITLE
Is it possible to show ArgsTable when React component returns null?

### DISCUSSION THREAD

**Discussion by marexandre** [storybookjs/storybook#15857] [Help] [answered]
Using storybook `v6.3.4` with React and have some custom components wrapping 3rd party lib that has nothing to do with React. Is it possible to get `ArgsTable` to show in docs page when a React components returns null?

React component code looks something like this:

```jsx
import coolLib from 'cool-lib';

export default function MyComponent({ id }) {
  useEffect(() => {
    const lib = new coolLib({ id });

    lib.on('load', () => {
      // do stuff
    });

    return () => {
      coolLib.remove()
    }
  }, []);

  return null;
}

MyComponent.propTypes = {
  id: PropTypes.string.isRequired
}
```

That `return null` causes `ArgsTable` to not show up in the DocsPage, if I return `<div>{id}</div>` or some other JSX the `ArgsTable` shows up...

**Comment by shilman** [comment, marked as answer]
We use `react-docgen` to generate the table, so your best bet is to see whether it support this option!
